0-dim arrays/vectors raise assertion errors

Following code raises assertion errors

```
import stan
import numpy as np

np.random.seed(26)

xdata = np.linspace(0, 50, 100)
b0, b1, sigma = -2, 1, 3
ydata = np.random.normal(loc=b1 * xdata + b0, scale=sigma)

refit_lr_code = """
data {
  // Define data for fitting
  int<lower=0> N;
  vector[N] x;
  vector[N] y;
  // Define excluded data. It will not be used when fitting.
  int<lower=0> N_ex;
  vector[N_ex] x_ex;
  vector[N_ex] y_ex;
}

parameters {
  real b0;
  real b1;
  real<lower=0> sigma_e;
}

model {
  b0 ~ normal(0, 10);
  b1 ~ normal(0, 10);
  sigma_e ~ normal(0, 10);
  for (i in 1:N) {
    y[i] ~ normal(b0 + b1 * x[i], sigma_e);  // use only data for fitting
  }
  
}

generated quantities {
    vector[N] log_lik;
    vector[N_ex] log_lik_ex;
    vector[N] y_hat;
    
    for (i in 1:N) {
        // calculate log likelihood and posterior predictive, there are 
        // no restrictions on adding more generated quantities
        log_lik[i] = normal_lpdf(y[i] | b0 + b1 * x[i], sigma_e);
        y_hat[i] = normal_rng(b0 + b1 * x[i], sigma_e);
    }
    for (j in 1:N_ex) {
        // calculate the log likelihood of the exluded data given data_for_fitting
        log_lik_ex[j] = normal_lpdf(y_ex[j] | b0 + b1 * x_ex[j], sigma_e);
    }
}
"""

data_dict = {
    "N": len(ydata),
    "y": ydata,
    "x": xdata,
    # No excluded data in initial fit
    "N_ex": 0,
    "x_ex": [],
    "y_ex": [],
}
sm = stan.build(program_code=refit_lr_code, data=data_dict)
```

```
Building...Error handling request
Traceback (most recent call last):
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/httpstan/views.py", line 277, in handle_show_params
    assert constrained_names, constrained_names
AssertionError: ()

Error handling request
Traceback (most recent call last):
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/home/ahartikainen/miniconda3/envs/stan/lib/python3.8/site-packages/httpstan/views.py", line 277, in handle_show_params
    assert constrained_names, constrained_names
AssertionError: ()
```

Adding print to `httpstan/views.py` I can see that there is empty tuple

```
('b0',)
('b1',)
('sigma_e',)
('log_lik.1', 'log_lik.2', 'log_lik.3', 'log_lik.4', 'log_lik.5', 'log_lik.6', 'log_lik.7', 'log_lik.8', 'log_lik.9', 'log_lik.10', 'log_lik.11', 'log_lik.12', 'log_lik.13', 'log_lik.14', 'log_lik.15', 'log_lik.16', 'log_lik.17', 'log_lik.18', 'log_lik.19', 'log_lik.20', 'log_lik.21', 'log_lik.22', 'log_lik.23', 'log_lik.24', 'log_lik.25', 'log_lik.26', 'log_lik.27', 'log_lik.28', 'log_lik.29', 'log_lik.30', 'log_lik.31', 'log_lik.32', 'log_lik.33', 'log_lik.34', 'log_lik.35', 'log_lik.36', 'log_lik.37', 'log_lik.38', 'log_lik.39', 'log_lik.40', 'log_lik.41', 'log_lik.42', 'log_lik.43', 'log_lik.44', 'log_lik.45', 'log_lik.46', 'log_lik.47', 'log_lik.48', 'log_lik.49', 'log_lik.50', 'log_lik.51', 'log_lik.52', 'log_lik.53', 'log_lik.54', 'log_lik.55', 'log_lik.56', 'log_lik.57', 'log_lik.58', 'log_lik.59', 'log_lik.60', 'log_lik.61', 'log_lik.62', 'log_lik.63', 'log_lik.64', 'log_lik.65', 'log_lik.66', 'log_lik.67', 'log_lik.68', 'log_lik.69', 'log_lik.70', 'log_lik.71', 'log_lik.72', 'log_lik.73', 'log_lik.74', 'log_lik.75', 'log_lik.76', 'log_lik.77', 'log_lik.78', 'log_lik.79', 'log_lik.80', 'log_lik.81', 'log_lik.82', 'log_lik.83', 'log_lik.84', 'log_lik.85', 'log_lik.86', 'log_lik.87', 'log_lik.88', 'log_lik.89', 'log_lik.90', 'log_lik.91', 'log_lik.92', 'log_lik.93', 'log_lik.94', 'log_lik.95', 'log_lik.96', 'log_lik.97', 'log_lik.98', 'log_lik.99', 'log_lik.100')
()
('b0',)
('b1',)
('sigma_e',)
('log_lik.1', 'log_lik.2', 'log_lik.3', 'log_lik.4', 'log_lik.5', 'log_lik.6', 'log_lik.7', 'log_lik.8', 'log_lik.9', 'log_lik.10', 'log_lik.11', 'log_lik.12', 'log_lik.13', 'log_lik.14', 'log_lik.15', 'log_lik.16', 'log_lik.17', 'log_lik.18', 'log_lik.19', 'log_lik.20', 'log_lik.21', 'log_lik.22', 'log_lik.23', 'log_lik.24', 'log_lik.25', 'log_lik.26', 'log_lik.27', 'log_lik.28', 'log_lik.29', 'log_lik.30', 'log_lik.31', 'log_lik.32', 'log_lik.33', 'log_lik.34', 'log_lik.35', 'log_lik.36', 'log_lik.37', 'log_lik.38', 'log_lik.39', 'log_lik.40', 'log_lik.41', 'log_lik.42', 'log_lik.43', 'log_lik.44', 'log_lik.45', 'log_lik.46', 'log_lik.47', 'log_lik.48', 'log_lik.49', 'log_lik.50', 'log_lik.51', 'log_lik.52', 'log_lik.53', 'log_lik.54', 'log_lik.55', 'log_lik.56', 'log_lik.57', 'log_lik.58', 'log_lik.59', 'log_lik.60', 'log_lik.61', 'log_lik.62', 'log_lik.63', 'log_lik.64', 'log_lik.65', 'log_lik.66', 'log_lik.67', 'log_lik.68', 'log_lik.69', 'log_lik.70', 'log_lik.71', 'log_lik.72', 'log_lik.73', 'log_lik.74', 'log_lik.75', 'log_lik.76', 'log_lik.77', 'log_lik.78', 'log_lik.79', 'log_lik.80', 'log_lik.81', 'log_lik.82', 'log_lik.83', 'log_lik.84', 'log_lik.85', 'log_lik.86', 'log_lik.87', 'log_lik.88', 'log_lik.89', 'log_lik.90', 'log_lik.91', 'log_lik.92', 'log_lik.93', 'log_lik.94', 'log_lik.95', 'log_lik.96', 'log_lik.97', 'log_lik.98', 'log_lik.99', 'log_lik.100')
()
```

Also, not sure why it does this 2 times before failing.

Final exception raised is

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-d508f7bb6005> in <module>
      8     "y_ex": [],
      9 }
---> 10 sm = stan.build(program_code=refit_lr_code, data=data_dict)
     11 #sample_kwargs = {"num_samples": 1000, "num_chains": 4}
     12 #fit = sm.sample(**sample_kwargs)

~/miniconda3/envs/stan/lib/python3.8/site-packages/stan/model.py in build(program_code, data, random_seed)
    424 
    425     try:
--> 426         return asyncio.run(go())
    427     except KeyboardInterrupt:
    428         return  # type: ignore

~/miniconda3/envs/stan/lib/python3.8/site-packages/nest_asyncio.py in run(future, debug)
     30         loop = asyncio.get_event_loop()
     31         loop.set_debug(debug)
---> 32         return loop.run_until_complete(future)
     33 
     34     if sys.version_info >= (3, 6, 0):

~/miniconda3/envs/stan/lib/python3.8/site-packages/nest_asyncio.py in run_until_complete(self, future)
     68                 raise RuntimeError(
     69                     'Event loop stopped before Future completed.')
---> 70             return f.result()
     71 
     72     def _run_once(self):

~/miniconda3/envs/stan/lib/python3.8/asyncio/futures.py in result(self)
    176         self.__log_traceback = False
    177         if self._exception is not None:
--> 178             raise self._exception
    179         return self._result
    180 

~/miniconda3/envs/stan/lib/python3.8/asyncio/tasks.py in __step(***failed resolving arguments***)
    278                 # We use the `send` method directly, because coroutines
    279                 # don't have `__iter__` and `__next__` methods.
--> 280                 result = coro.send(None)
    281             else:
    282                 result = coro.throw(exc)

~/miniconda3/envs/stan/lib/python3.8/site-packages/stan/model.py in go()
    413             resp = await client.post(f"/{model_name}/params", json={"data": data})
    414             if resp.status != 200:
--> 415                 raise RuntimeError(resp.json()["message"])
    416             params_list = resp.json()["params"]
    417             assert len({param["name"] for param in params_list}) == len(params_list)

~/miniconda3/envs/stan/lib/python3.8/site-packages/stan/common.py in json(self)
     22 
     23     def json(self) -> dict:
---> 24         return simdjson.loads(self.content)
     25 
     26 

~/miniconda3/envs/stan/lib/python3.8/site-packages/simdjson/__init__.py in loads(s, cls, object_hook, parse_float, parse_int, parse_constant, object_pairs_hook, **kwargs)
     59 
     60     parser = Parser()
---> 61     return parser.parse(s, True)
     62 
     63 

ValueError: The JSON document has an improper structure: missing or superfluous commas, braces, missing keys, etc.

```

Versions (both from PyPI):
httpstan: '4.3.2'
pystan: '3.0.0rc1'

--- a/httpstan/views.py
+++ b/httpstan/views.py
@@ -272,8 +272,6 @@ async def handle_show_params(request: aiohttp.web.Request) -> aiohttp.web.Respon
     params = []
     for name, dims_ in zip(param_names, dims):
         constrained_names = tuple(filter(lambda s: re.match(fr"^{name}\.\S+|^{name}\Z", s), constrained_param_names))
-        assert isinstance(dims_, list)
-        assert constrained_names, constrained_names
         params.append(schemas.Parameter().load({"name": name, "dims": dims_, "constrained_names": constrained_names}))
     return aiohttp.web.json_response({"name": model_name, "params": params})
 

--- a/tests/test_fits.py
+++ b/tests/test_fits.py
@@ -10,6 +10,7 @@ import helpers
 
 headers = {"content-type": "application/json"}
 program_code = "parameters {real y;} model {y ~ normal(0, 0.0001);}"
+program_code_vector = "parameters {vector[2] z; vector[0] x;} model {z ~ normal(0, 0.0001);}"
 
 
 @pytest.mark.asyncio
@@ -68,3 +69,27 @@ async def test_fits_random_seed(api_url: str) -> None:
     # look at all draws
     assert np.allclose(draws1, draws2)
     assert not np.allclose(draws1, draws3)
+
+
+@pytest.mark.asyncio
+async def test_fits_vector_sizes(api_url: str) -> None:
+    """Simple test of sampling with zero and non-zero vector sizes."""
+
+    num_samples = num_warmup = 500
+    payload = {
+        "function": "stan::services::sample::hmc_nuts_diag_e_adapt",
+        "random_seed": 123,
+        "num_samples": num_samples,
+        "num_warmup": num_warmup,
+    }
+    param_name = "z.1"
+    draws = await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)
+    assert len(draws) == num_samples, (len(draws), num_samples)
+
+    param_name = "z.2"
+    draws = await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)
+    assert len(draws) == num_samples, (len(draws), num_samples)
+
+    param_name = "x.1"
+    with pytest.raises(KeyError, match="No draws found for parameter `x.1`."):
+        await helpers.sample_then_extract(api_url, program_code_vector, payload, param_name)

